araproc/analysis/standard_reco.py

```diff
@@ -924,3 +926,346 @@ def get_avg_corr_snr(self, wave_packet, excluded_channels = []):
+            wavepacket, polarization, channels_to_csw, reference_ch, 
+            arrival_delays_reco, wf_type, which_distance)  
+
+        # Initialize the final CSW waveform time and voltage arrays
+        csw_values = np.zeros((1, csw_length))
+        csw_times = self.__trim_array(
+            np.asarray(wavepacket['waveforms'][reference_ch].GetX()), 
+            np.asarray(wavepacket['waveforms'][reference_ch].GetY()), 
+            wavepacket['waveforms'][reference_ch].GetN()-csw_length
+        )[0]
+        csw_dt = csw_times[1] - csw_times[0]
+
+        # Roll the waveform from each channel so the starting time of each
+        #   waveform lines up. Then add the waveform to the CSW.
+        for ch_ID in channels_to_csw: 
+
+            # Load this channel's voltage and time arrays. Shift time by arrival delay
+            values = np.asarray(wavepacket['waveforms'][ch_ID].GetY())
+            times = np.asarray(wavepacket['waveforms'][ch_ID].GetX()) - arrival_delays[ch_ID]
+
+            # Determine if the time binning of this channel matches the time binning
+            #   of the CSW and adjust if it doesn't. 
+            # Ex: If the CSW has times of [0.9, 1.4, 1.9] then a channel
+            #   with a time array of [1.6, 2.1, 2.6] needs to be rebinned/shifted 
+            #   backwards by 0.2 but a channel with the time array [3.4, 3.9, 4.4 ] 
+            #   doesn't because each bin starts and ends on the same fraction of a 
+            #   nanosecond that the CSW does. 
+            rebinning_shift = round( (times[0] - csw_times[0]) % csw_dt, 4) 
+            if rebinning_shift != 0: 
+                waveform_model = Akima1DInterpolator(times, values)
+                times = times - rebinning_shift
+                values = waveform_model(times)
+
+            # Trim this waveform's length to match the CSW length but try 
+            #   not to remove the maximal waveform point
+            # TODO could combine this logic with the rolling
+            times, values = self.__trim_array(times, values, len(values)-csw_length)
+
+            # Roll the waveform so that the start and end times of the waveform
+            #   line up exactly with the CSW
+            roll_shift = (times[0] - csw_times[0]) / csw_dt
+            # TODO Check that roll_shift is close to an integer
+            # TODO Check that the peak of the waveform or the expected signal 
+            #   bin isn't about to get rolled 
+            rolled_wf = np.roll( values, int(roll_shift) )
+            rolled_times = np.linspace(
+                times[0] - roll_shift*(times[1] - times[0]),
+                times[-1] - roll_shift*(times[1] - times[0]),
+                len(times)
+            )
+
+            # Add this channel's waveform to the CSW
+            csw_values = np.nansum( np.dstack( (csw_values[0], rolled_wf) ), axis=2) 
+
+        # Un-nest the csw. csw.shape was (1,len(big_times)) but is now len(big_times)
```

Comment:
I think `big_times` doesn't exist anymore? Also maybe `np.squeeze` is clearer?

Reply:
You're totally right about `big_times`. I have not heard of `np.sqeeze` but I can switch it